testing/e2e/app/routes/Breadcrumbs/spec.ts

```diff
@@ -0,0 +1,94 @@
+import { test, expect, Page } from '@playwright/test';
+
+test.describe('Breadcrumbs', () => {
+  test(`should overflow whenever there is not enough space`, async ({
+    page,
+  }) => {
+    await page.goto(`/Breadcrumbs`);
+
+    const setContainerSize = getSetContainerSize(page);
+    const expectOverflowState = getExpectOverflowState(page);
+
+    await expectOverflowState({
+      expectedItemLength: 5,
+      expectedOverflowButtonVisibleCount: undefined,
+    });
+
+    await setContainerSize('200px');
+
+    await expectOverflowState({
+      expectedItemLength: 2,
+      expectedOverflowButtonVisibleCount: 2,
+    });
+
+    // should restore hidden items when space is available again
+    await setContainerSize(undefined);
+
+    await expectOverflowState({
+      expectedItemLength: 5,
+      expectedOverflowButtonVisibleCount: undefined,
+    });
+  });
+
+  test(`should at minimum always show one overflow tag and one item`, async ({
+    page,
+  }) => {
+    await page.goto(`/Breadcrumbs`);
+
+    const setContainerSize = getSetContainerSize(page);
+    const expectOverflowState = getExpectOverflowState(page);
+
+    await expectOverflowState({
+      expectedItemLength: 5,
+      expectedOverflowButtonVisibleCount: undefined,
+    });
+
+    await setContainerSize('10px');
+
+    await expectOverflowState({
+      expectedItemLength: 1,
+      expectedOverflowButtonVisibleCount: 1,
+    });
+  });
+});
+
+// ----------------------------------------------------------------------------
+
+const getSetContainerSize = (page: Page) => {
```

Comment:
optional: one layer of abstraction could be removed by simply creating a `setContainerSize` that accepts both the `page` object and the actual value.

same comment applies to `expectOverflowState` and any other similar utils in other test scenarios